internal/gengapic/client_init.go

```diff
@@ -135,155 +135,238 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
-		p("}")
-		p("")
+	g.imports[pbinfo.ImportSpec{Path: "google.golang.org/grpc"}] = true
+	g.imports[pbinfo.ImportSpec{Path: "google.golang.org/grpc/metadata"}] = true
 
-		g.imports[pbinfo.ImportSpec{Path: "google.golang.org/grpc"}] = true
-		g.imports[pbinfo.ImportSpec{Path: "google.golang.org/grpc/metadata"}] = true
-	}
+	g.grpcClientUtilities(serv, servName, imp, hasLRO)
+}
 
-	// Client constructor
-	{
-		clientName := camelToSnake(serv.GetName())
-		clientName = strings.Replace(clientName, "_", " ", -1)
-
-		p("// New%sClient creates a new %s client.", servName, clientName)
-		p("//")
-		g.comment(g.comments[serv])
-		p("func New%[1]sClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sClient, error) {", servName)
-		p("  clientOpts := default%sClientOptions()", servName)
-		p("")
-		p("  if new%sClientHook != nil {", servName)
-		p("    hookOpts, err := new%sClientHook(ctx, clientHookParams{})", servName)
-		p("    if err != nil {")
-		p("      return nil, err")
-		p("    }")
-		p("    clientOpts = append(clientOpts, hookOpts...)")
-		p("  }")
-		p("")
-		p("  disableDeadlines, err := checkDisableDeadlines()")
-		p("  if err != nil {")
-		p("    return nil, err")
-		p("  }")
-		p("")
-		p("  connPool, err := gtransport.DialPool(ctx, append(clientOpts, opts...)...)")
+func (g *generator) grpcClientUtilities(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasLRO bool) {
+	p := g.printf
+
+	clientName := camelToSnake(serv.GetName())
+	clientName = strings.Replace(clientName, "_", " ", -1)
+
+	// Factory function
+	p("// New%sClient creates a new %s client.", servName, clientName)
+	p("//")
+	g.comment(g.comments[serv])
+	p("func New%[1]sClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sClient, error) {", servName)
+	p("  clientOpts := default%sClientOptions()", servName)
+	p("")
+	p("  if new%sClientHook != nil {", servName)
+	p("    hookOpts, err := new%sClientHook(ctx, clientHookParams{})", servName)
+	p("    if err != nil {")
+	p("      return nil, err")
+	p("    }")
+	p("    clientOpts = append(clientOpts, hookOpts...)")
+	p("  }")
+	p("")
+	p("  disableDeadlines, err := checkDisableDeadlines()")
+	p("  if err != nil {")
+	p("    return nil, err")
+	p("  }")
+	p("")
+	p("  connPool, err := gtransport.DialPool(ctx, append(clientOpts, opts...)...)")
+	p("  if err != nil {")
+	p("    return nil, err")
+	p("  }")
+	p("  c := &%sClient{", servName)
+	p("    connPool:    connPool,")
+	p("    disableDeadlines: disableDeadlines,")
+	p("    CallOptions: default%sCallOptions(),", servName)
+	p("")
+	p("    %s: %s.New%sClient(connPool),", grpcClientField(servName), imp.Name, serv.GetName())
+	p("  }")
+	p("  c.setGoogleClientInfo()")
+	p("")
+
+	if hasLRO {
+		p("  c.LROClient, err = lroauto.NewOperationsClient(ctx, gtransport.WithConnPool(connPool))")
 		p("  if err != nil {")
+		p("    // This error \"should not happen\", since we are just reusing old connection pool")
```

Comment:
For my Golang edification: When no variables are needed, what's the motivation for using single-line string values instead of a multi-line literal?

Reply:
This is just the convention we've been using in the Go generator and in Showcase since the beginning. If we were starting from scratch, I'd certainly support multi-line literals, especially if we used them with the dedent package. Given the convention, though, I suggest not changing things right now. (Possibly in a separate, dedicated PR, if we decide it's worth the effort.)

BTW, multi-line literals can have variables, though since they're not labeled inline it gets confusing fast when the format specifier is far removed from its corresponding parameter.